unity/JetBrains.Rider.Unity.Editor/EditorPlugin/AssetPostprocessors/CsprojAssetPostprocessor.cs

```diff
@@ -332,54 +333,89 @@ private static void ApplyCustomReference(string name, XElement projectContentEle
     // Set appropriate version
     private static void FixTargetFrameworkVersion(XElement projectElement, XNamespace xmlns)
     {
-        var targetFrameworkVersion = projectElement.Elements(xmlns + "PropertyGroup")
-          .Elements(xmlns + "TargetFrameworkVersion")
-          .FirstOrDefault(); // Processing csproj files, which are not Unity-generated JetBrains/Unity3dRider#56
-        if (targetFrameworkVersion == null)
-          return;
-
-      if (UnityUtils.ScriptingRuntime > 0)
+      SetOrUpdateProperty(projectElement, xmlns, "TargetFrameworkVersion", s =>
         {
-          var version = PluginSettings.TargetFrameworkVersion;
-          if (!PluginSettings.OverrideTargetFrameworkVersion)
+          string version = string.Empty;
+          try
           {
+            version = s.Substring(1);
```

Comment:
This will throw an exception if the string is less than 1 char long - so if there's no `TargetFrameworkVersion` element, or it's empty, we'll get an exception.

Reply:
it is under try-cath. I added additional check for null or empty.